collectors/cgroups.plugin/sys_fs_cgroup.c

```diff
@@ -73,6 +73,129 @@ static uint32_t Write_hash = 0;
 static uint32_t user_hash = 0;
 static uint32_t system_hash = 0;
 
+enum cgroups_type { CGROUPS_AUTODETECT_FAIL, CGROUPS_V1, CGROUPS_V2 };
+
+enum cgroups_systemd_setting {
+    SYSTEMD_CGROUP_ERR,
+    SYSTEMD_CGROUP_LEGACY,
+    SYSTEMD_CGROUP_HYBRID,
+    SYSTEMD_CGROUP_UNIFIED
+};
+
+struct cgroups_systemd_config_setting {
+    char *name;
+    enum cgroups_systemd_setting setting;
+};
+
+static struct cgroups_systemd_config_setting cgroups_systemd_options[] = {
+    { .name = "legacy",  .setting = SYSTEMD_CGROUP_LEGACY  },
+    { .name = "hybrid",  .setting = SYSTEMD_CGROUP_HYBRID  },
+    { .name = "unified", .setting = SYSTEMD_CGROUP_UNIFIED },
+    { .name = NULL,      .setting = SYSTEMD_CGROUP_ERR     },
+};
+
+/* on Fed systemd is not in PATH for some reason */
+#define SYSTEMD_CMD_RHEL "/usr/lib/systemd/systemd --version"
+#define SYSTEMD_HIERARCHY_STRING "default-hierarchy="
+
+#define MAXSIZE_PROC_CMDLINE 4096
+static enum cgroups_systemd_setting cgroups_detect_systemd(const char *exec)
+{
+    pid_t command_pid;
+    enum cgroups_systemd_setting retval = SYSTEMD_CGROUP_ERR;
+    char buf[MAXSIZE_PROC_CMDLINE];
```

Comment:
In normal situation we are not creating buffers with the same size that we give as argument, please add more 1 character to `buf`, to avoid it has the same size given to `fgets`

Reply:
Hi, according to `fgets` docu:
```
This is the maximum number of characters to be read (including the final null-character). Usually, the length of the array passed as str is used.
``

Reply:
I understand this, and this was the motive I said Netdata is always appending an additional character as we can see on claim:

https://github.com/netdata/netdata/blob/13624215508c3bd472cb9ac5e9665df4f5006167/claim/claim.c#L60

Your code is not wrong, but we are doing different on other files.

Reply:
I do not see a point in allocating one extra byte that never will get used.

Reply:
I see your point, as I said I was worried only with the fat we would have two patterns in our code.